components/remote_settings/src/lib.rs

```diff
@@ -24,6 +28,48 @@ use storage::Storage;
 
 uniffi::setup_scaffolding!("remote_settings");
 
+/// The `AppContext` object represents the parameters and characteristics of the
+/// consuming application. For `remote-settings`, it is used to filter locally stored `records`.
+/// We always fetch all `records` from the remote-settings storage. Some records could have a `filter_expression`
+/// attached to them, which will be matched against the `AppContext`. So just `records` for this particualar instance
+/// will be used if set.
```

Comment:
> So just `records` for this particualar instance will be used if set.

I would rephrase into something simpler, for example: When set, only records where the expression is true will be returned.